.gitlab-ci.yml

```
@@ -51,7 +51,7 @@ build:workspace:
         .
     - echo "IMAGE_VERSION=$CI_COMMIT_BRANCH" >> image_version.env
   rules:
-    - if: $CI_COMMIT_BRANCH == "main"
+    - if: $CI_COMMIT_BRANCH == "main" && $CI_PIPELINE_SOURCE != "pipeline"
```

For easier reading and future extensions, I recommend one rule per line, like:
```suggestion
    - if: '$CI_PIPELINE_SOURCE == "pipeline"'
      when: never
    - if: $CI_COMMIT_BRANCH == "main"
```

---

I thought about this. but if we do it the job will run on every PR in mender-mcu-integration.

---

Isn't that what we want, to not run it on triggers (like mender-mcu tiggering this repo)?

`pipeline` applies only to multi-project pipelines. See [documentation](https://docs.gitlab.com/ee/ci/jobs/job_rules.html#ci_pipeline_source-predefined-variable). For other cases like schedules or just regular push to main it will run.

---

You pointed out when I created the trigger that we might want to specify which branch of mender-mcu-integration we want, which likely will happen. That might mean that mender-mcu-integration has made changes that requires an updated workspace, and we will only get that if we trigger the workspace job - since that both builds and specifies which image the build jobs should use.

I guess we can also say that that is not possible - and for changes to the workspace, we can rather specify the revision of mender-mcu here.

---

@danielskinstad I apologize, I miss-read the if condition in the first place.

Thank you for the f2f discussion - it looks good!